docs/architecture/adr-050-sign-mode-textual.md

```diff
@@ -129,6 +130,44 @@ Msg (<int>/<int>): <string>           // E.g. Msg (1/2): bank v1beta1 send coins
 End of transaction messages
 ```
 
+### 9. Custom `Msg` Renderers
+
+Application developers may choose to not follow default renderer value output for their own `Msg`s. In this case, they can implement their own custom `Msg` renderer. This is similar to [EIP4430](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-4430.md), where the smart contract developer chooses the description string to be shown to the end user.
+
+This is done by setting the `cosmos.msg.v1.textual.custom_renderer` Protobuf option to a non-empty string. This option CAN ONLY be set on a Protobuf message representing transaction message object (implementing `sdk.Msg` interface).
+
+```proto
+message MsgFooBar {
+  // Optional comments to describe in human-readable language the formatting
+  // rules of the custom renderer.
+  option (cosmos.msg.v1.textual.custom_renderer) = "<unique algorithm identifier>";
```

Comment:
Given the discussions, I'm inclined to do:

```suggestion
  option (cosmos.msg.v1.textual.unsafe_custom_renderer) = "<unique algorithm identifier>";
```

ref: https://github.com/cosmos/cosmos-sdk/pull/12785#discussion_r937012747

WDYT?

Reply:
TBH, I don't think `unsafe*` API should be exposed. It doesn't look good.

Reply:
Agreed about "unsafe" - maybe 'expert" instead? Or just go with your original thought and make it a boolean. You'll want to register the renderer anyhow, so the algorithm identifier doesn't seem to be adding much.

Reply:
We agreed on our call to name this "expert"